File: general-interview-prep/javascript/reorderLogFiles.js
/**
 * @param {string[]} logs
 * @return {string[]}
 */
 var reorderLogFiles = function(logs) {
    let [digits, letters] = [[], []];

    for(let i = 0; i < logs.length; i++) {
        const str = logs[i].slice(logs[i].indexOf(' ') + 1);
        if(str.slice(0,1).match(/[a-z]/i)) {
            letters.push(logs[i]);
        } else {
            digits.push(logs[i]);
        }
    }


    const sortedLets = letters.sort((a,b) => {
        const idA = a.split(" ",1);
        const idB = b.split(" ",1);
        const contentA = a.split(/^[^\s]*\s/);
        const contentB = b.split(/^[^\s]*\s/)

        if(contentA[1] !== contentB[1]) return contentA[1].localeCompare(contentB[1]); 
        else return idA[0].localeCompare(idB[0]); 

    });
    sortedLets.push(...digits);
    return sortedLets;
};

File: general-interview-prep/golang/reorderLogFiles.go
import (
	"regexp"
	"strings"
)

func reorderLogFiles(logs []string) []string {
	alphaTest := regexp.MustCompile("[A-Za-z]")
	alphaLogs := make([]string, 0)
	digiLogs := make([]string, 0)
	answer := make([]string, 0)

	for _, log := range logs {
			strLog := string(log)
			splitLog := strings.Join(strings.Split(strLog, " ")[1:], "")
			found := alphaTest.FindAllString(splitLog, -1)
			
			if len(found) > 0 {
					alphaLogs = append(alphaLogs, strLog)
			} else {
					digiLogs = append(digiLogs, strLog)
			}
	}
	
	sort.Slice(alphaLogs, func (i, j int) bool {
			iIndex := strings.Index(alphaLogs[i], " ")
			jIndex := strings.Index(alphaLogs[j], " ")
			
			iLog := alphaLogs[i][iIndex+1:]
			jLog := alphaLogs[j][jIndex+1:]
			if iLog == jLog {
					return alphaLogs[i] < alphaLogs[j]
			}
			return iLog < jLog
	})
	
	
	answer = append(answer, alphaLogs...)
	answer = append(answer, digiLogs...)
	
	return answer
}